modules/apps/app-builder/app-builder-web/src/main/resources/META-INF/resources/js/pages/apps/edit/EditApp.es.js

```diff
@@ -85,11 +85,61 @@ export default ({
 		title = Liferay.Language.get('edit-app');
 	}
 
-	const getEmptyState = (description, title) => {
-		return {
-			description,
-			title,
-		};
+	const stepType = ['DATA_LAYOUT', 'DATA_LIST_VIEW', 'WORKFLOW_PROCESS'];
+
+	const stepProps = {
+		DATA_LAYOUT: {
+			emptyState: {
+				description: Liferay.Language.get(
+					'create-one-or-more-forms-to-display-the-data-held-in-your-data-object'
+				),
+				title: Liferay.Language.get('there-are-no-form-views-yet'),
+			},
+			endpoint: `/o/data-engine/v2.0/data-definitions/${dataDefinitionId}/data-layouts`,
+			title: (
+				<StepTitle title={Liferay.Language.get('select-a-form-view')} />
+			),
+		},
+		DATA_LIST_VIEW: {
+			emptyState: {
+				description: Liferay.Language.get(
+					'create-one-or-more-tables-to-display-the-data-held-in-your-data-object'
+				),
+				title: Liferay.Language.get('there-are-no-table-views-yet'),
+			},
+			endpoint: `/o/data-engine/v2.0/data-definitions/${dataDefinitionId}/data-list-views`,
+			title: (
+				<StepTitle
+					title={Liferay.Language.get('select-a-table-view')}
+				/>
+			),
+		},
+		WORKFLOW_PROCESS: {
+			emptyState: {
+				search: {
+					className: 'taglib-search-state',
+					title: Liferay.Language.get('no-results-were-found'),
+				},
+			},
+			endpoint: `/o/headless-admin-workflow/v1.0/workflow-definitions?page=-1&pageSize=-1`,
```

Comment:
@gabrielpramos We must pass the `active` parameter to this endpoint to just retrieve the active definitions.

Thx

Reply:
Ok, I've adjusted it when I rebased, I'll update the PR
thx :D